feat: add wallet page

lujakob opened the pull request:
### Describe the changes you have made in this PR

Add wallet page and update menu accordingly. Combine "receive", "send" and "transactions" into one single "wallet" page.

### Link this PR to an issue [optional]

Fixes #2243

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

<img width="1203" alt="Bildschirmfoto 2023-04-09 um 14 00 57" src="https://user-images.githubusercontent.com/886152/230792075-6677df15-66bd-41d0-a738-f2c38b0a5bde.png">

### How has this been tested?

Manually

### Checklist

- [ ] My code follows the style guidelines of this project and performed a self-review of my own code
- [ ] New and existing tests pass locally with my changes
- [ ] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

lujakob commented:
This is a draft to be able to have a first look.

I copied the `<BalanceBox>` from [PR](https://github.com/getAlby/lightning-browser-extension/pull/2318) to be able to test it. As soon as this PR is merged to master I will overwrite the component.

The "see all transactions" link is not implemented yet. It is not available on transactions page yet either. I think this should be done in a follow up PR. (actually there are issues already regarding this)

bitcoinuser commented:
Hi,

If in all Alby places the name is "Account", why choose "Wallet" word? Thanks.

reneaaron commented:
@lujakob I just merged #2236 :raised_hands: Feel free to continue with this one :wink:

lujakob commented:
> @lujakob I just merged #2236 🙌 Feel free to continue with this one 😉

Yesss, awesome. I just noticed and am back on it 😀

lujakob commented:
@reneaaron I think this is ready for review as far as I can see now. Please have a look. 

I did not change the active state color of the tabs header links "Outgoing" / "Incoming" as seen on the design, because I didn't know which color/class it is and I didn't find an example in the UI.

I don't have transactions in my Alby testnet accounts. Sending/receiving didn't work. (I wrote you a Telegram message)

lujakob commented:
@reneaaron I tested incoming/outgoing transactions with different lnbits accounts now and it works fine. Both are filtered by account now. From my point of view only the tab menu item active state color is missing now.

reneaaron commented:
Awesome, thanks very much for your PR! Just did some first round of testing and worked great so far. :+1: 

> I did not change the active state color of the tabs header links "Outgoing" / "Incoming" as seen on the design, because I didn't know which color/class it is and I didn't find an example in the UI.

I think that's fine for now. We're currently working on new tab styles in #2240.

I'd prefer to keep the balance box on top of the buttons to better align this view with the popup. Possibly we need to limit the button size (they seem very huge to me on desktop), but we can do that in a followup PR. cc @stackingsaunter 
 
![image](https://user-images.githubusercontent.com/100827540/235518434-b21ff71e-61f9-4214-807a-ac7de759698e.png)

You now removed the Transactions page. Wasn't the idea to link to this page with a `See all` link where we could display more of them (or at one day offer pagination / incremental loading)?

lujakob commented:
@reneaaron 

> I'd prefer to keep the balance box on top of the buttons to better align this view with the popup. Possibly we need to limit the button size (they seem very huge to me on desktop), but we can do that in a followup PR. cc @stackingsaunter

I moved the balance box on top.

> You now removed the Transactions page. Wasn't the idea to link to this page with a `See all` link where we could display more of them (or at one day offer pagination / incremental loading)?

True. I didn't have this in mind anymore. I reverted the Transactions page component. In the wallet page the list is limited to 5 items. "See all transactions" navigates to `/transactions/incoming` or `/transactions/outgoing` according to the selected tab. I abstracted the fetching of transactions and invoices into hooks, as they are used in both wallet and transactions. The components look a lot cleaner now as well.

lujakob commented on src/app/router/Options/Options.tsx:
```diff
@@ -55,7 +56,17 @@ function Options() {
             <Route path="receive" element={<Receive />} />
-            <Route path="transactions" element={<Transactions />} />
+            <Route path="wallet" element={<Wallet />} />
+            <Route path="transactions">
```
@reneaaron I tried to redirect the page to the first child in case of a user typing in the "/transactions" url manually to prevent showing a white page. I did not succeed. Maybe you have an idea.

reneaaron replied:
I don't think we need to do that.

lujakob commented on src/app/screens/Transactions/index.tsx:
```diff
@@ -55,17 +49,17 @@ function Transactions() {
         <div>
-          {transactions?.length > 0 && (
-            <TransactionsTable transactions={transactions} />
+          {Array.isArray(list) && list.length > 0 && (
```
fyi: I had to add Array.isArray for type reasons here.

lujakob commented on src/extension/background-script/actions/accounts/info.ts:
```diff
@@ -7,10 +7,8 @@ const info = async (message: MessageAccountInfo) => {
-    connector.getInfo(),
-    connector.getBalance(),
-  ]);
+  const info = await connector.getInfo();
```
This is a debugging leftover. Good catch. I will revert it.

reneaaron reviewed (changes requested):
I think we should merge the `Wallet` and the  `DefaultView` into one screen as there is quite some code duplication which already produces bugs (eg the one page displays outgoing transactions from all accounts, while the other does only display transactions from the current account, tab styles don't match, etc). 

Clarified with @stackingsaunter to remove the "Wallet" heading and add the links "See all transactions" to the `DefaultView`. 

I think the safest way forward is to remove the new Wallet page (but keep the new hooks for loading transactions) and change the existing `DefaultView` to use the new logic and display the "see all" links.

Does that make sense to you?

reneaaron commented:
Yes, I think that's the idea. Very much like e.g. we also don't have 2 different receive screens for the popup / full view, they are just used on those different pages.

lujakob commented:
> Yes, I think that's the idea. Very much like e.g. we also don't have 2 different receive screens for the popup / full view, they are just used on those different pages.

I refactored the `DefaultView` component accordingly and removed the `Wallet` component. Please have another look.

FYI: I added the `isOptionsPage` prop to be able to show the "Wallet" title on the options page. Initially I used a Tailwind CSS condition, then I noticed it doesn't work, because we always want to show the title on the options page, also for narrow screens.

reneaaron commented:
Thanks very much for the update, that is a lot better now already. :+1: 

However, I think we should get rid of most (if not all) differences between the two different views. (they add quite some complexity for no good reason IMO)  

I would: 
 - Remove the heading "Wallet" (also for the options page) 
 - Have the same page size (10 is okay, also for the popup view, everything > 4 will be below the fold anyway and I don't think that's a problem) 
 - ~~Open the "see all transaction" links in the same context (don't open a new options page if you click them in the popup)~~ I see the problem now, that would require us to handle those pages differently too, as they would need to have back buttons in the popup. Let's skip that for now and open the options page like you implemented it. :+1: 

Some UI related things I noticed: 
 - Do we want to add the wallet link to the dropdown here? It's just opening the same page really. If we do, I think we should add a proper wallet icon
![image](https://github.com/getAlby/lightning-browser-extension/assets/100827540/44d10f79-8ccf-4418-9e21-67ebc3cd6c91)
 - The "see all transactions" links appear visually too separated to me (the tabs are closer to the transactions as the links in the bottom which makes me think this is an unrelated link)
![image](https://github.com/getAlby/lightning-browser-extension/assets/100827540/66e79aa0-ed35-405e-bfe1-8c338905aa80)
 - The buttons are imo too huge (never seen this big buttons anywhere before 😅)
 
 I'll check back with @stackingsaunter what we could do regarding the UI related things.

lujakob commented:
> * Remove the heading "Wallet" (also for the options page)

I removed it

> * Have the same page size (10 is okay, also for the popup view, everything > 4 will be below the fold anyway and I don't think that's a problem)

I don't understand what you mean by page size? I actually understand nothing of this sentence 😅
> * ~Open the "see all transaction" links in the same context (don't open a new options page if you click them in the popup)~ I see the problem now, that would require us to handle those pages differently too, as they would need to have back buttons in the popup. Let's skip that for now and open the options page like you implemented it. 👍
> 
> Some UI related things I noticed:
> 
> * Do we want to add the wallet link to the dropdown here? It's just opening the same page really. If we do, I think we should add a proper wallet icon

I removed it. I think Saunter also mentioned the same thing a while ago.

> * The "see all transactions" links appear visually too separated to me (the tabs are closer to the transactions as the links in the bottom which makes me think this is an unrelated link)
I adjusted the distance
> * The buttons are imo too huge (never seen this big buttons anywhere before 😅)
> 
> I'll check back with @stackingsaunter what we could do regarding the UI related things.

Yes let me know about it

lujakob commented:
@reneaaron @stackingsaunter any updates on this one? How would you like to modify the buttons?

reneaaron commented:
Fixed some minor things and tested everything. Thanks very much @lujakob for the PR, great work! :muscle: 

tACK

lujakob commented:
@reneaaron thanks for testing and doing the final updates. I merged master.

lujakob commented:
@reneaaron can we merge this? I have two dependency tickets waiting for this. Thx.
